src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -211,7 +211,7 @@ void InputModel::InputModelTFImpl::load_places() {
                 OPENVINO_DEBUG("TensorFlow Frontend: Placeholder ", op_name, " does not have 'shape' attribute");
             }
             auto output_shapes_any = node_decoder->get_attribute("_output_shapes");
-            if (pshape.rank().is_static() && pshape.rank().get_length() == 0 &&
+            if ((pshape.rank().is_dynamic() || pshape.rank().get_length() == 0) &&
```

Comment:
I have uploaded a test model for this case.

Reply:
$ **ovc yz_header.pb** 
[ ERROR ]  -------------------------------------------------
[ ERROR ]  ----------------- INTERNAL ERROR ----------------
[ ERROR ]  Unexpected exception happened.
[ ERROR ]  Please verify parameters and environment.
[ ERROR ]  If you think this is a bug, please create new ticket here: 
[ ERROR ]  https://github.com/openvinotoolkit/openvino/issues.
[ ERROR ]  -------------- DETAILED INFORMATION -------------
[ ERROR ]  Check 'data_rank.compatible(filters_rank)' failed at src/core/shape_inference/include/convolution_shape_inference_util.hpp:298:
While validating node 'opset1::Convolution resnet_model/conv2d/Conv2D (opset1::Transpose resnet_model/Pad[0]:f32[], opset1::Transpose Transpose_24[0]:f32[64,3,7,7]) -> (f32[?,64,1..,1..])' with friendly_name 'resnet_model/conv2d/Conv2D':
Data batch and filters rank do not match (data batch shape: [], filters shape: [64,3,7,7]).

[ ERROR ]  Traceback (most recent call last):
  File "/home/sgui/miniforge3/envs/p38/lib/python3.8/site-packages/openvino/tools/ovc/convert_impl.py", line 494, in _convert
    ov_model = driver(argv, {"conversion_parameters": non_default_params})
  File "/home/sgui/miniforge3/envs/p38/lib/python3.8/site-packages/openvino/tools/ovc/convert_impl.py", line 250, in driver
    ov_model = moc_emit_ir(prepare_ir(argv), argv)
  File "/home/sgui/miniforge3/envs/p38/lib/python3.8/site-packages/openvino/tools/ovc/convert_impl.py", line 194, in prepare_ir
    ov_model = moc_pipeline(argv, moc_front_end)
  File "/home/sgui/miniforge3/envs/p38/lib/python3.8/site-packages/openvino/tools/ovc/moc_frontend/pipeline.py", line 296, in moc_pipeline
    ov_model = moc_front_end.convert(input_model)
  File "/home/sgui/miniforge3/envs/p38/lib/python3.8/site-packages/openvino/frontend/frontend.py", line 18, in convert
    converted_model = super().convert(model)
RuntimeError: Check 'data_rank.compatible(filters_rank)' failed at src/core/shape_inference/include/convolution_shape_inference_util.hpp:298:
While validating node 'opset1::Convolution resnet_model/conv2d/Conv2D (opset1::Transpose resnet_model/Pad[0]:f32[], opset1::Transpose Transpose_24[0]:f32[64,3,7,7]) -> (f32[?,64,1..,1..])' with friendly_name 'resnet_model/conv2d/Conv2D':
Data batch and filters rank do not match (data batch shape: [], filters shape: [64,3,7,7]).


[ ERROR ]  ----------------- END OF REPORT -----------------
[ ERROR ]  -------------------------------------------------

Reply:
I think the real problem is in a different part. This fix can reveal the real problem. That because we should work with dynamic rank parameter in conversion and finally it will deduce parameter's rank relying on bottom operations.

Thanks for the reproducer. I will take a look where the problem. Ideally, we should use `_output_shape` because this is not Placeholder attribute according to TF specification.

Best regards,
Roman